ginkgo.Jenkinsfile

```diff
@@ -54,66 +53,173 @@ pipeline {
                }
             }
         }
-        stage('Boot VMs'){
+        stage ("copy-code-and-boot-vms"){
             options {
                 timeout(time: 30, unit: 'MINUTES')
             }
+
             environment {
-                TESTDIR="${WORKSPACE}/${PROJ_PATH}/test"
+                FAILFAST=setIfLabel("ci/fail-fast", "true", "false")
+                CONTAINER_RUNTIME=setIfLabel("area/containerd", "containerd", "docker")
             }
-            steps {
-                sh 'cd ${TESTDIR}; NETNEXT=true K8S_VERSION=1.10 vagrant up --no-provision'
-                sh 'cd ${TESTDIR}; K8S_VERSION=1.14 vagrant up --no-provision'
+            parallel {
+                stage('boot-vms-runtime') {
+                    environment {
+                        GOPATH="${WORKSPACE}/runtime-gopath"
+                        TESTDIR="${GOPATH}/${PROJ_PATH}/test"
+                    }
+                    steps {
+                        sh 'mkdir -p ${GOPATH}/src/github.com/cilium'
+                        sh 'cp -a ${WORKSPACE}/${PROJ_PATH} ${GOPATH}/${PROJ_PATH}'
+                        sh 'cd ${TESTDIR}; vagrant up runtime --provision'
+                    }
+                    post {
+                        failure {
+                            script {
+                                if  (currentBuild.description == '') {
+                                    currentBuild.description = 'runtime vm provisioning fail'
+                                }
+                            }
+                        }
+                    }
+                }
+                stage('boot-vms-K8s-1.10-net-next') {
+                    environment {
+                        GOPATH="${WORKSPACE}/k8s-lower-gopath"
```

Comment:
Is there any chance we can set K8s_VERSION above this and use it as part of the directory name. It's really unclear what this is (I'm guessing it's the lower/older version)

Reply:
Makes sense